platform/mbed_wait_api.h

```diff
@@ -60,6 +62,10 @@ extern "C" {
  *    microsecond precision up to 10 milliseconds. If delay is larger than 10 milliseconds and not in ISR, it is the same as
  *    `wait_ms`. We recommend `wait_us` and `wait_ms` over `wait`.
  */
+MBED_DEPRECATED_SINCE("mbed-os-5.13",
+                      "'wait'-named functions are now reserved for CPU-using delays that are safe to call from IRQ context. "
+                      "This call should be replaced by either a sleep call such as ThisThread::sleep_for (C++) or "
+                      "thread_sleep_for (C), or by a wait call such as wait_us(), depending on required behaviour.")
```

Comment:
This message is a bit confusing to me. Is the goal to eventually remove `wait()` and `wait_ms()` with `wait_us()` only remaining as busy-wait implementation? Or will `wait()` and `wait_ms()` eventually transition to being a busy-wait implementation as well?

Reply:
Goal is to remove `wait` and `wait_ms` - people should use `ThisThread::sleep` instead. `wait_us` and `wait_ns` remain as busy-waits, and appropriate for odd-context use.

Other functions will sleep and are thread-context only. (`ThisThread::sleep`, `Mutex::lock`, `Semaphore::acquire`).

Aim is to reduce confusion about what these calls do, by name adjustment.

Now, we don't quite get there because there are still a couple of "wait"s in the RTOS API - `ConditionVariable::wait` and `EventFlags::wait`, and they obviously sleep, but I don't think there's a better name; cv::wait is totally C/C++ standard and I'd rather not deviate from CMSIS-RTOS either. (`Semaphore::wait` is there now, but I'm intending to make that become `Semaphore::acquire`.

This might be an argument for shifting `wait_us` and `wait_ns` to a new name like `delay_us` and `delay_ns`.  (Would also align us to the Linux kernel, which uses `sleep` and `delay` as its two terms...)

Reply:
Would it hurt to be more explicit? ```delay_execution_ns``` , ```delay_blocking_ns``` ,  ```no_operation_ns```

Reply:
None of those are immediately grabbing me. "Blocking" often refers to sleeping. "Execution" is saying what? And there aren't necessarily any no-ops involved. Maybe you could include "spin" or "busy"?

Reply:
If I had to make a suggestion for the deprecation message, I'd go with something similar to following. I think it gets to the resolution a bit quicker.

```c++

MBED_DEPRECATED_SINCE("mbed-os-5.13",
                      "'wait' is deprecated in favor of explicit sleep functions. To sleep, 'wait' should be replaced by "
                      "'ThisThread::sleep_for' (C++) or 'thread_sleep_for' (C). If you wish to wait (without sleeping), call "
                      "'wait_us'. 'wait_us' is safe to call from IRQ context.")
```